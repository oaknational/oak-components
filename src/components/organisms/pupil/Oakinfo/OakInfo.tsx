import React, { MouseEventHandler, ReactNode, useState } from "react";

import { OakInfoButton } from "../OakinfoButton";

import { OakTooltip, OakTooltipProps } from "@/components/molecules";

export type OakInfoProps = {
  /**
   * Some content to give as a hint to answer a question
   */
  hint: ReactNode;
  isLoading?: boolean;
  disabled?: boolean;
} & Omit<OakTooltipProps, "children" | "tooltip" | "id">;

/**
 * Presents a button which will show a hint when clicked
 */
export const OakInfo = (props: OakInfoProps) => {
  const { hint, isLoading, disabled, ...tooltipProps } = props;
  const [isOpen, setIsOpen] = useState(false);
  const handleClick: MouseEventHandler = () => {
    setIsOpen(!isOpen);
  };

  return (
    <OakTooltip
      tooltip={props.hint}
      isOpen={isOpen}
      id={"info-tooltip"}
      {...tooltipProps}
    >
      <OakInfoButton onClick={handleClick} isOpen={isOpen} />
    </OakTooltip>
  );
};
