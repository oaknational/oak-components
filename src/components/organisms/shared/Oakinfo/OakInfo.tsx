import React, { MouseEventHandler, ReactNode, useState } from "react";

import { OakInfoButton } from "@/components/organisms/shared/OakinfoButton";
import { OakTooltip, OakTooltipProps } from "@/components/molecules";
import { OakBox } from "@/components/atoms";

export type OakInfoProps = {
  /**
   * Some content to give as a hint to answer a question
   */
  hint: ReactNode;
  id: string;
  isLoading?: boolean;
  disabled?: boolean;
} & Omit<OakTooltipProps, "children" | "tooltip" | "id">;

/**
 * Presents a button which will show a hint when clicked
 */
export const OakInfo = (props: OakInfoProps) => {
  const { hint, id, isLoading, disabled, ...tooltipProps } = props;
  const [isOpen, setIsOpen] = useState(false);
  const handleClick: MouseEventHandler = () => {
    setIsOpen(!isOpen);
  };

  return (
    <>
      <OakBox $display="none" id={id}>
        {hint}
      </OakBox>
      <OakTooltip tooltip={hint} isOpen={isOpen} {...tooltipProps}>
        <OakInfoButton
          onClick={handleClick}
          isOpen={isOpen}
          buttonProps={{ "aria-describedby": id }}
        />
      </OakTooltip>
    </>
  );
};
