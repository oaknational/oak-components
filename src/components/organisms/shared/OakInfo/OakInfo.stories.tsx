import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import { OakInfo } from "./OakInfo";

import { OakBox } from "@/components/atoms";

const meta: Meta<typeof OakInfo> = {
  component: OakInfo,
  tags: ["autodocs"],

  argTypes: {
    hint: {
      control: {
        type: "text",
      },
    },
    tooltipPosition: {
      options: ["top-right", "bottom-left", "top-left", "top-right"],
    },
  },
  decorators: [
    (Story) => (
      <OakBox $mt="space-between-xxl">
        <Story />
      </OakBox>
    ),
  ],
  args: {
    hint: "We've put the lessons in order helping you build on what you've learned before so itâ€™s best to start with the first lesson of a unit.",
    tooltipPosition: "top-left",
  },
};
export default meta;

type Story = StoryObj<typeof OakInfo>;

export const Default: Story = {
  render: (args) => <OakInfo {...args} />,
};
