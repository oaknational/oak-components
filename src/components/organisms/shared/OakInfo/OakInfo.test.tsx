import React from "react";
import "@testing-library/jest-dom";

import { OakInfo } from "./OakInfo";

import renderWithTheme from "@/test-helpers/renderWithTheme";
import {
  installMockIntersectionObserver,
  installMockResizeObserver,
} from "@/test-helpers";

installMockIntersectionObserver();
installMockResizeObserver();

describe("OakInfo", () => {
  it("matches snapshot", () => {
    const { container } = renderWithTheme(
      <OakInfo
        hint="The answer is right in front of your eyes"
        id="info-tooltip"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
