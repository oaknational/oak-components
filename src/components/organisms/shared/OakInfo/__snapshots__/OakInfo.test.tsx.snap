// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakInfo matches snapshot 1`] = `
.c0 {
  display: none;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c1 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c7 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c9 {
  position: relative;
  width: 2.5rem;
  min-width: 2.5rem;
  height: 2.5rem;
  color: #222222;
  background: #ffe555;
  border-radius: 6.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c11 {
  position: absolute;
  top: 0rem;
  width: 100%;
  height: 100%;
  border-radius: 6.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c12 {
  position: relative;
  width: 2rem;
  min-width: 2rem;
  height: 2rem;
  min-height: 2rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c13 {
  -webkit-filter: invert(10%) sepia(1%) saturate(236%) hue-rotate(314deg) brightness(95%) contrast(91%);
  filter: invert(10%) sepia(1%) saturate(236%) hue-rotate(314deg) brightness(95%) contrast(91%);
  object-fit: contain;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c14 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c5 {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  text-align: left;
  font-family: unset;
  outline: none;
  font-family: --var(google-font),Lexend,sans-serif;
  color: #222222;
}

.c5:disabled {
  pointer-events: none;
  cursor: default;
}

.c6 {
  display: inline-block;
  position: relative;
}

.c6:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #222222;
}

.c6:active {
  color: #222222;
}

.c6:disabled {
  color: #808080;
}

.c3 > :first-child:focus-visible .shadow {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%), 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c3 > :first-child:hover .shadow {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%);
}

.c3 > :first-child:active .shadow {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%), 0.25rem 0.25rem 0 rgba(87,87,87,100%);
}

.c3 > :first-child:disabled .icon-container {
  background: #808080;
}

.c3 > :first-child:hover .icon-container {
  background: #ffe555;
}

.c3 > :first-child:active .icon-container {
  background: #ffe555;
}

.c4:hover .shadow {
  box-shadow: none !important;
}

.c4:active .shadow {
  box-shadow: 0.125rem 0.125rem 0 rgba(255,229,85,100%), 0.25rem 0.25rem 0 rgba(87,87,87,100%) !important;
}

<div>
  <div
    class="c0"
    id="info-tooltip"
  >
    The answer is right in front of your eyes
  </div>
  <div
    style="display: contents;"
  >
    <div
      class="c1 c2 c3 c4"
    >
      <button
        aria-describedby="info-tooltip"
        aria-label="open info tooltip"
        class="c5 c6"
        data-rac="true"
      >
        <div
          class="c7 c8"
        >
          <div
            class="c9 c10 icon-container"
          >
            <div
              class="c11 shadow"
            />
            <div
              class="c12"
              data-icon-for="button"
            >
              <img
                alt=""
                class="c13"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1709052869/icons/Icon_Info_vsx3xi.svg"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
          </div>
          <span
            class="c14"
          />
        </div>
      </button>
    </div>
  </div>
</div>
`;
